.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,44 @@
+name: Build & Test
+
+on: [push, pull_request]
+
+env: 
+  # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
+  BUILD_TYPE: Release
```

Comment:
I think this `env` section isn't needed. I don't see anywhere the `BUILD_TYPE` environment variable is used.